tests/templates/test_embeddings/test_iqp_emb.py

```diff
@@ -0,0 +1,226 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for the IQPEmbedding template.
+"""
+import pytest
+import numpy as np
+import pennylane as qml
+from pennylane import numpy as pnp
+
+
+class TestDecomposition:
+    """Tests that the template defines the correct decomposition."""
+
+    QUEUES = [
+        (1, ["Hadamard", "RZ"], [[0], [0]]),
+        (2, ["Hadamard", "RZ", "Hadamard", "RZ", "MultiRZ"], [[0], [0], [1], [1], [0, 1]]),
+        (
+            3,
+            ["Hadamard", "RZ", "Hadamard", "RZ", "Hadamard", "RZ", "MultiRZ", "MultiRZ", "MultiRZ"],
+            [[0], [0], [1], [1], [2], [2], [0, 1], [0, 2], [1, 2]],
+        ),
+    ]
+
+    @pytest.mark.parametrize("n_wires, expected_names, expected_wires", QUEUES)
+    def test_expansion(self, n_wires, expected_names, expected_wires):
+        """Checks the queue for the default settings."""
+
+        features = list(range(n_wires))
+
+        op = qml.templates.IQPEmbedding(features, wires=range(n_wires))
+        tape = op.expand()
+
+        for i, gate in enumerate(tape.operations):
+            assert gate.name == expected_names[i]
+            assert gate.wires.labels == tuple(expected_wires[i])
+
```

Comment:
Would be nice to have a test for non-default settings too, e.g., for `n_repeats` and `pattern`.